source/code/plugins/oms_common.rb

```diff
@@ -891,7 +891,13 @@ def initialize(refresh_interval_seconds)
       @cache_lock = Mutex.new
       @refresh_interval_seconds = refresh_interval_seconds
       @condition = ConditionVariable.new
-      @thread = Thread.new(&method(:refresh_cache))
+      # Important: Disabling refresh functionality to prevent data loss btw rsyslog and OMS
+      # This bug happen because OMS is blocked 15 seconds due to DNS timeout when resolving 
```

Comment:
How would disabling the cache refresh/invalidation address the issue of DNS lookups timing out?

Reply:
During OMS start-up there would be some DNS looks timeout but these will be cached as null values. So if the same unkown hostname were sent later it won't cause OMS to timeout again. By removing the 5min refresh this will prevent OMS from being blocked and lose data on each 5min interval.

Reply:
Setting the timeout to 3 seconds may help. I have updated the PR.